flake.nix

```diff
@@ -26,6 +26,11 @@
           kmir-pyk = poetry2nix.mkPoetryApplication {
             python = nixpkgs-pyk.python310;
             projectDir = ./kmir;
+            src = rv-utils.lib.mkPykAppSrc {
+              pkgs = import nixpkgs { system = prev.system; };
+              src = ./kmir;
+              cleaner = poetry2nix.cleanPythonSources;
+            };
```

Comment:
Does this just clean stale data that might be around?

Reply:
Sort of; in order for Nix builds to be reproducible they usually define a cleaning step that removes files not relevant to the build. Here the reference to `cleanPythonSources` is a function that removes things like git ignored files, python compiled objects, etc.

Normally Poetry2Nix handles this for us transparently, but to work round an upstream bug we need to call a function in our central Nix tools repo that patches some of KMir's code to be compatible with Nix.